### PR TITLE
247 dectect station delay

### LINKED ISSUE #247: Detect Station Arrival Delay Operator
### I want
basicly a refactor of our main, but with tests

matched live and planned difference calcuation and sending of traindelaynotification

## Tasks
- [x] Open Pull Request
- [ ] Gain knowledge about different best practices
- [ ] Prepare a short presentation to share your insights with the Team

## Acceptance Criteria
- [x] Documented
- [ ] Merged in master
- [ ] Linter is okay
- [x] Tests are written and green
- [x] Code Review done
- [x] Reviewed by PO

### PR DESCRIPTION
**Pull request by skydivin4ng3l**
Refactors Delay detection based on Live and Planned Station arrival Data including tests

changes TrainDelayNotifcation location_id to station_id to match conventions of other events

fixes #247

### CODE REVIEW

File: core/src/main/java/org/bptlab/cepta/operators/SumOfDelayAtStationFunction.java [hunk 1]
```diff
@@ -52,7 +52,7 @@ public void process(Integer key, Context context, Iterable<TrainDelayNotificatio
                 HashMap<Long, Double> sums = new HashMap<Long, Double>();
                 for (TrainDelayNotification in: input) {
                     Long trainId = in.getTrainId();
-                    Long locationId = in.getLocationId();
+                    Long locationId = in.getStationId();
```
Comment (Quacck) [inline core/src/main/java/org/bptlab/cepta/operators/SumOfDelayAtStationFunction.java]: is this locationId or stationId?  I think other parts of our code use stationId as the variablename but im not sure

File: core/src/test/java/org/bptlab/cepta/detect-station-delay-tests/DetectStationArrivalDelayTest.java [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Iterator;
+import java.util.concurrent.atomic.AtomicBoolean;
+
+public class DetectStationArrivalDelayTest {
+    @Test
+    public void testDelayPositive() throws IOException {
+        AtomicBoolean pass = new AtomicBoolean(true);
```
Comment (Quacck) [inline core/src/test/java/org/bptlab/cepta/detect-station-delay-tests/DetectStationArrivalDelayTest.java]: Why do we use an AtomicBoolean here? We should either add a comment or just use a plain and simple boolean

File: core/src/test/java/org/bptlab/cepta/detect-station-delay-tests/DetectStationArrivalDelayTest.java [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+        //Duration delay = Duration.newBuilder().setSeconds(3600).build();
```
Comment (Quacck) [inline core/src/test/java/org/bptlab/cepta/detect-station-delay-tests/DetectStationArrivalDelayTest.java]: Is this comment still needed?

File: core/src/test/java/org/bptlab/cepta/providers/LiveTrainDataProvider.java [hunk 4]
```diff
@@ -39,6 +39,9 @@ public static LiveTrainData getDefaultLiveTrainDataEvent() {
     return builder.build();
   }
 
+
+
+
```
Comment (Quacck) [inline core/src/test/java/org/bptlab/cepta/providers/LiveTrainDataProvider.java]: I guess we could remove these lines to be like the others

File: core/src/test/java/org/bptlab/cepta/providers/LiveTrainDataProvider.java [hunk 5]
```diff
@@ -39,6 +39,9 @@ public static LiveTrainData getDefaultLiveTrainDataEvent() {
...
   // @DataProvider(name = "live-train-data-provider")
```
Comment (Quacck) [inline core/src/test/java/org/bptlab/cepta/providers/LiveTrainDataProvider.java]: Is this comment still up-to-date?

File: core/src/main/java/org/bptlab/cepta/operators/SumOfDelayAtStationFunction.java (hunk 1 above)
Comment (skydivin4ng3l) [reply core/src/main/java/org/bptlab/cepta/operators/SumOfDelayAtStationFunction.java]: This change should solve all LocationID to StationID so there should be no LocationID anymore

File: core/src/test/java/org/bptlab/cepta/detect-station-delay-tests/DetectStationArrivalDelayTest.java (hunk 2 above)
Comment (skydivin4ng3l) [reply core/src/test/java/org/bptlab/cepta/detect-station-delay-tests/DetectStationArrivalDelayTest.java]: We need it for the ForEach

File: core/src/test/java/org/bptlab/cepta/providers/LiveTrainDataProvider.java (hunk 5 above)
Comment (skydivin4ng3l) [reply core/src/test/java/org/bptlab/cepta/providers/LiveTrainDataProvider.java]: do don't know

**Pull request merged**